File: DSA/VikrantSinghBanwala_590018082/day13_question1.c
#include<stdio.h>
#include<string.h>
#include<ctype.h>
void cntWrdFindLgst(char *str, int *wordCount, char *longestWord) {
    char *token;
    *wordCount = 0;
    longestWord[0] = '\0';
    int maxLength = 0;
    token = strtok(str, " \t\n");
    while (token != NULL) {
        (*wordCount)++;
        int length = strlen(token);
        if (length > maxLength) {
            maxLength = length;
            strcpy(longestWord, token);
        }
        token = strtok(NULL, " \t\n");
    }
}
int main() {
    char str1[] = "Hello world programming";
    int count1;
    char longest1[100];
    cntWrdFindLgst(str1, &count1, longest1);
    printf("Word count: %d, Longest word: \"%s\"\n", count1, longest1);

    char str2[] = " Java Python C++ ";
    int count2;
    char longest2[100];
    cntWrdFindLgst(str2, &count2, longest2);
    printf("Word count: %d, Longest word: \"%s\"\n", count2, longest2);
    return 0;
}

File: DSA/pranavdixit_590014418/Day_13_Question_1.py
def word_count_and_longest(s):
    words = s.strip().split()
    word_count = len(words)
    longest_word = max(words, key=len) if words else ""
    print("Word count:", word_count)
    print("Longest word:", f'"{longest_word}"')

word_count_and_longest("Hello world programming")
word_count_and_longest(" Java Python C++ ")
word_count_and_longest("a bb ccc dddd")
